crates/jstz_proto/src/runtime/v2/fetch/http.rs

```
@@ -22,7 +22,12 @@ pub struct Response {
 
 impl Into<http::Response<Option<Vec<u8>>>> for Response {
     fn into(self) -> http::Response<Option<Vec<u8>>> {
-        let mut builder = http::Response::builder().status(self.status);
+        // According to JavaScript documentation, `Response.error()` returns a response with status code 0
+        // and is mainly used for client side network errors. In regular JS, the fetch promise would be rejected.
+        // Within Jstz, the smart function can only return this if it explicitly called `Response.error()`, which
+        // means the intent is closer to a 400 Bad Request.
+        let status = if self.status == 0 { 400 } else { self.status };
```

Similarly, I don't think we should implicitly change the status code here. I agree that the user probably intended something different, but we should be compliant with spec behaviour of `Response.error`

---

`http` won't allow a 0 status code

---

Ah, because the spec doesn't consider 0 as a status code. Hmm, I think implicitly rewriting the status code instead of failing here a bit odd -- but I think as long as we make this explicit in docs, then we should be fine. Ironically Deno just fails silently relying on https://github.com/denoland/deno/blob/f54afe03bd5902576a1e8e71d0473a63279a663a/ext/fetch/23_response.js#L180, but this doesn't handle `Response.error`